File: R/8kyu/8kyu-well-of-ideas-easy-version.r
# Well of Ideas - Easy Version
# Language: R
# Version 3.4.1
# Difficulty: 8 kyu

well <- function(x){
  good = length(x[which(x == "good")])
  
  if (good > 0 & good <= 2){
    "Publish!"
  }else if (good > 2){
    "I smell a series!"
  }else{
    "Fail!"
  }
}


# Tests
test_that('Basic tests', {
  expect_equal(well(c('bad', 'bad', 'bad')), 'Fail!')
  expect_equal(well(c('good', 'bad', 'bad', 'bad', 'bad')), 'Publish!')
  expect_equal(well(c('good', 'bad', 'bad', 'bad', 'bad', 'good', 'bad', 'bad', 'good')), 'I smell a series!')
})


ifelse(!'good'%in%x,'Fail!',ifelse(sum(x=='good')>2,'I smell a series!','Publish!'))

File: Python/8kyu/8kyu-well-of-ideas-easy-version.py
# Well of Ideas - Easy Version
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

# Solution 1
def well(x):
    good = x.count("good")
    
    if good > 0 and good <= 2:
        return "Publish!"
    elif good > 2:
        return "I smell a series!"
    else:
        return "Fail!"

# Improved Solution 1 to follow logical order
def well(x):
    good = x.count("good")

    if good == 0:
        return "Fail!"
    elif good <= 2:
        return "Publish!"
    else:
        return "I smell a series!"

# Tests
import codewars_test as test
from solution import well

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(well(['bad', 'bad', 'bad']), 'Fail!')
        test.assert_equals(well(['good', 'bad', 'bad', 'bad', 'bad']), 'Publish!')
        test.assert_equals(well(['good', 'bad', 'bad', 'bad', 'bad', 'good', 'bad', 'bad', 'good']), 'I smell a series!')